x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -53,11 +79,103 @@ public class IlmHealthIndicatorService implements HealthIndicatorService {
     );
 
     public static final String AUTOMATION_DISABLED_IMPACT_ID = "automation_disabled";
+    public static final String STAGNATING_INDEX_IMPACT_ID = "stagnating_index";
+    public static final List<HealthIndicatorImpact> AUTOMATION_DISABLED_IMPACT = List.of(
+        new HealthIndicatorImpact(
+            NAME,
+            AUTOMATION_DISABLED_IMPACT_ID,
+            3,
+            "Automatic index lifecycle and data retention management is disabled. The performance and stability of the cluster "
+                + "could be impacted.",
+            List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)
+        )
+    );
+
+    public static final List<HealthIndicatorImpact> STAGNATING_INDEX_IMPACT = List.of(
+        new HealthIndicatorImpact(
+            NAME,
+            STAGNATING_INDEX_IMPACT_ID,
+            3,
+            "Automatic index lifecycle and data retention management cannot make progress on one or more indices. The performance and "
+                + "stability of the indices and/or the cluster could be impacted.",
+            List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)
+        )
+    );
+
+    private static final TimeValue ONE_DAY = TimeValue.timeValueDays(1);
+
+    static final Map<String, RuleConfig> RULES_BY_ACTION_CONFIG = Map.of(
+        RolloverAction.NAME,
+        actionRule(RolloverAction.NAME).stepRules(
+            stepRule(WaitForActiveShardsStep.NAME, ONE_DAY),
+            stepRule(WaitForRolloverReadyStep.NAME, ONE_DAY),
+            stepRule(RolloverStep.NAME, ONE_DAY)
```

Comment:
I believe we should tack number of retries for these ones (as they're async steps that move into the step from ERROR when they're retried)

Please update them to track max 100 retries

Reply:
the number of retries is tracked by default in the stepRules

Reply:
Ah, yes I see that.

> We should not track time in WaitForRolloverReadyStep as we could be there for days and that's normal

I think having it mandatory AND implicit is a bit confusing, especially for things where we maybe don't want to track time like this one above ^^ ( where we'd ONLY want to track number of retries) 

Should we have it mandatory to specify one of maxTimeOn or number of retries? (with the option to specify both as well- all options explicit in the constructors)

Reply:
tackled on #96855 96855